### PR TITLE
Requiring python 3.7.11 or newer

### PR DESCRIPTION
**Pull request by acpaquette**
Original PR #4547

# Task List

Most of the tasks for this PR involve transitioning ISIS from using ASC maintained feedstocks to those on the conda forge. 
For the most part this requires specifying newer versions of dependencies that are available on conda-forge, but some feedstocks from https://github.com/USGS-Astrogeology/isis3_dependencies/ may have to be submitted as-is to the conda-forge staged-recipes project. Mostly this involves updating the yaml files to specify to use the conda-forge channel for each dependency and to avoid the defaults channel. 

Kakadu can't be moved to conda-forge, but since it doesn't depend on much/anything likely no action is needed.

Given that conda-forge is using a much more modern version of GCC than v4, it may cause some issues simultaneously supporting older centos versions. 

- [x] TNT
  - [x] add TNT to conda forge https://github.com/conda-forge/tnt-feedstock
  - [x] add ASC maintainer
- [x] JAMA
  - [x] add JAMA to conda forge  https://github.com/conda-forge/jama-feedstock
  - [x] add ASC maintainer
- [x] GMM 
  - [x] add GMM to conda forge https://github.com/conda-forge/gmm-feedstock
  - [x] add ASC maintainer
- [ ] Bullet
  - [ ] @KrisBecker's PR to patch/refactor bullet usage in ISIS
  - [ ] make bullet build variant with Double Precision https://github.com/conda-forge/bullet-feedstock/issues/39  
  - [ ] add ASC maintainer
- [x] OpenCV
  - [x] upgrade to version 3.4.7 should be modern enough for python versions younger than 3.6
- [x] embree
  - [x] switch to conda-forge
- [x] qhull
  - [x] switch to conda-forge
- [x] qt 
  - [x] switch to conda-forge
- [x] qwt
  - [x] switch to conda-forge
- [x] xerces-c
  - [x] switch to conda-forge
- [x] x264
  - [x] update to newer build on conda-forge
- [x] jpeg
  - [x] update to newer build on conda-forge
- [x] Update recipe/meta.yml to be inline with environment.yml 

## Description
add python as an explicit dependency in as many yaml files as I could find

## Related Issue
#3885 

## How Has This Been Tested?
not tested because of boost issues and lack of publicly accessible test data

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
@acpaquette Just closed #3566. We are good to remove building support for GCC4. Maybe we should update the text on the PR with respect to GCC4. The issue is that building with GCC is going to be problematic. Running the built code on a CentOS7 system that has gcc4 as the default compiler should not be an issue at all because of how the dependencies are encapsulated and provided via conda.

**Comment by jessemapel** [comment]
Fixs #4656

File: isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp
Comment (jessemapel) [inline isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:143]: Is SURF no longer supported?

File: recipe/meta.yaml
Comment (jessemapel) [inline recipe/meta.yaml]: This needs to be changed back before we merge

File: isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp
Comment (acpaquette) [reply isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:143]: Correct, OpenCV moved SURF out and into an [OpenCV Contrib repo](https://github.com/opencv/opencv_contrib) where we would have to build OpenCV ourselves against the contrib library to maintain that functionality.

I'm guessing this constitutes an API breaking change :/

Comment (jessemapel) [reply isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:143]: I think this is likely API breaking.

At a minimum, we need to update the findfeatures docs so that SURF isn't listed in them.

Comment (Kelvinrr) [reply isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:143]: @jessemapel Should I simply remove them from the algorithm list or remove examples that imply you can use it as well?

Comment (KrisBecker) [reply isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:143]: If I may suggest, perhaps there is way to retain all the code, trap the error and gracefully continue. This because if we (or other developers for that matter) can ever get SURF back, no change would need to be made to the code. findfeatures should rightfully throw an error if its requested and cannot find/invoke the algorithm.

Comment (Kelvinrr) [reply isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:143]: @KrisBecker So you're suggestion is leave this commented and let OpenCV throw it's error? Presumably we would still have to change the docs to remove the parameter?

**Review by jessemapel — DISMISSED** [review]

**Comment by jessemapel** [comment]
I don't think there's a lot of value in leaving the SIFT stuff in and allowing it to error. We can always pull the files out of the VCS if we get it back in somehow.

**Comment by Kelvinrr** [comment]
@jessemapel I must of accidentally dismissed your last review since I pushed the changlog change shortly after, does this look good?

**Review by jessemapel — DISMISSED** [review]

**Review by jessemapel — APPROVED** [review]

**Pull request merged**
